jobs/release.Jenkinsfile

```diff
@@ -152,34 +152,64 @@ lock(resource: "release-${params.STREAM}", extra: locks) {
             }
         }
 
-        stage("Push OSContainer Manifest") {
-            // Ship a manifest list containing all requested architectures.
-            def oscontainer_registry_repo = pipecfg.registry_repos?.oscontainer
-            if (oscontainer_registry_repo) {
-                withCredentials([file(variable: 'REGISTRY_SECRET',
-                                      credentialsId: 'oscontainer-push-registry-secret')]) {
-                    def arch_args = basearches.collect{"--arch ${it}"}.join(" ")
-                    shwrap("""
-                    cosa push-container-manifest --auth=\${REGISTRY_SECRET} \
-                        --repo=${oscontainer_registry_repo} --tag=${params.STREAM} \
-                        --artifact=ostree --metajsonname=base-oscontainer \
-                        --build=${params.VERSION} ${arch_args}
-                    """)
-                }
-            }
-            def oscontainer_old_registry_repo = pipecfg.registry_repos?.oscontainer_old
-            if (oscontainer_old_registry_repo) {
-                // For a period of time let's also mirror into the old location too
-                // Drop this after October 2022. See
-                // https://discussion.fedoraproject.org/t/updated-registry-location-for-fedora-coreos-ostree-native-container/42740
-                withCredentials([file(credentialsId: 'oscontainer-secret', variable: 'REGISTRY_SECRET')]) {
-                    shwrap("""
-                    skopeo copy --all --authfile \$REGISTRY_SECRET \
-                        docker://${oscontainer_registry_repo}:${params.STREAM} \
-                        docker://${oscontainer_old_registry_repo}:${params.STREAM}
-                    """)
-                }
-            }
+        def registry_repos = pipeutils.get_registry_repos(pipecfg, params.STREAM)
+
+        // [config.yaml name -> [meta.json 'images' name, meta.json toplevel name, tag suffix]]
+        def push_containers = ['oscontainer': ['ostree', 'base-oscontainer', ''],
+                               'extensions': ['extensions-container', 'extensions-container', 'extensions'],
+                               'legacy_oscontainer': ['ostree', 'oscontainer', 'legacy']]
+
+        // filter out those not defined in the config
+        push_containers.keySet().retainAll(registry_repos.keySet())
+
+        // filter out those not built. this step makes the assumption that if an
+        // image isn't built for x86_64, then we don't upload it at all
+        def artifacts = readJSON(text: shwrapCapture("cosa meta --dump")).images.keySet()
+        // in newer Groovy, retainAll can take a closure, which would be nicer here
```

Comment:
Looks like it works for collections but not maps. And yup, compare the "Since" in https://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/Collection.html#retainAll(groovy.lang.Closure) vs https://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/Map.html#retainAll(groovy.lang.Closure).